### PR TITLE
:allowed_clock_drift should be bidrectional

### PR DESCRIPTION
**Pull request by johnnyshields**
Fixes https://github.com/onelogin/ruby-saml/issues/599

:allowed_clock_drift should be bidrectional (allow X sec before "NotBefore" and X sec after "NotOnOrAfter"). Also improves readability of associated error messages.

### CODE REVIEW

**Comment by pitbulk** [comment]
Can you add some unittest to cover the cases?. It seems this change has not impacted the test suite and it should, so we may be missing some unitest here

**Comment by johnnyshields** [comment]
@sronsiek could you help write tests for this fix?

**Comment by johnnyshields** [comment]
@pitbulk specs done, ready for review

**Comment by pitbulk** [comment]
@johnnyshields the specs fail

**Comment by johnnyshields** [comment]
@pitbulk this can be merged. the one failing run is not related to this PR

**Comment by johnnyshields** [comment]
@pitbulk let's merge?

**Comment by pitbulk** [comment]
- Can you clarify why the Float::EPSILON is required?

**Comment by johnnyshields** [comment]
Yes, floats have built in inaccuracy (the uncertainty factor is "epsilon") which can cause <= to return false when it should be true.

My code extends the clock drift factor by epsilon which errs on the side of being permissive when comparing exact times.

**Review by pitbulk — APPROVED** [review]

File: lib/onelogin/ruby-saml/response.rb [hunk 1]
```diff
@@ -337,9 +337,9 @@ def audiences
       end
 
       # returns the allowed clock drift on timing validation
-      # @return [Integer]
+      # @return [Float]
       def allowed_clock_drift
-        return options[:allowed_clock_drift].to_f
+        options[:allowed_clock_drift].to_f.abs + Float::EPSILON
```
[the remark this answers is not in the document]
Comment (johnnyshields) [reply lib/onelogin/ruby-saml/response.rb:342]: Yes, floats have built in inaccuracy (the uncertainty factor is "epsilon") which can cause <= to return false when it should be true.

My code extends the clock drift factor by epsilon which errs on the side of being permissive when comparing exact times.

**Pull request merged**